build/source/dshare/flxMapping.f90

```diff
@@ -0,0 +1,205 @@
+ flux2state_orig(iLookFLUX%scalarExfiltration)              = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%scalarSurfaceRunoff)             = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%mLayerSatHydCondMP)              = flux2state(state1=integerMissing,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%mLayerSatHydCond)                = flux2state(state1=integerMissing,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%iLayerSatHydCond)                = flux2state(state1=integerMissing,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%mLayerHydCond)                   = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%iLayerLiqFluxSoil)               = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%mLayerLiqFluxSoil)               = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%mLayerBaseflow)                  = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%mLayerColumnInflow)              = flux2state(state1=integerMissing,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%mLayerColumnOutflow)             = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%scalarSoilBaseflow)              = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%scalarSoilDrainage)              = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%scalarAquiferRecharge)           = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%scalarAquiferTranspire)          = flux2state(state1=iname_matLayer,  state2=integerMissing)
+ flux2state_orig(iLookFLUX%scalarAquiferBaseflow)           = flux2state(state1=iname_matLayer,  state2=integerMissing)
+
+ ! ** copy across flux metadata
+ do iVar=1,size(flux_meta)
+  flux2state_orig(iVar)%var_info = flux_meta(iVar)
+ end do
+
+ ! ** check all variables are defined
+ do iVar=1,size(flux_meta)
+  if(flux2state_orig(iVar)%state1==integerUndefined .or. flux2state_orig(iVar)%state2==integerUndefined)then
+   message=trim(message)//'flux-to-state mapping is undefined for variable "'//trim(flux_meta(iVar)%varname)//'"'
+   err=20; return
+  endif
+ end do
+
+ ! -----
+ ! - liquid water state variables...
+ ! ---------------------------------
+
+ ! initialize to the original structure
+ do iVar=1,size(flux_meta)
+  flux2state_liq(iVar)%state1 = flux2state_orig(iVar)%state1 
+  flux2state_liq(iVar)%state2 = flux2state_orig(iVar)%state2
+ end do
+
+ ! modify the state type names associated with the flux mapping structure
+ do iVar=1,size(flux_meta)
+  ! (mass of total water on the vegetation canopy --> mass of liquid water)
+  if(flux2state_liq(iVar)%state1==iname_watCanopy) flux2state_liq(iVar)%state1=iname_liqCanopy
```

Comment:
It could work.. though I'm not sure about the efficiency gains for a collection of `where` statements as opposed to `if` statements within a loop. Since the benefits of the change are unclear, I'll keep the code as it is for now.